### PR TITLE
fix: login required before action

### PR DESCRIPTION
**Pull request by dudu0506**
https://mask.atlassian.net/browse/FW-1726

# Pull Request

## Related Issue

Closes FW-0000

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

File: src/components/Actions/ArticleMore.tsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ export const ArticleMoreAction = memo<MoreProps>(function ArticleMoreAction({ ar
     const identity = author.handle || ens || formatEthereumAddress(author.id, 4);
     return (
         <MoreActionMenu
+            mustLogin
```
Comment (guanbinrui) [inline src/components/Actions/ArticleMore.tsx:33]: No need to add this prop, cause login is required for this actions by default.

File: src/components/MoreActionMenu.tsx [hunk 2]
```diff
@@ -29,8 +36,10 @@ export function MoreActionMenu({ disabled, button, children, className, onClick
                 aria-label="More"
                 onClick={async (event) => {
                     event.stopPropagation();
-                    if (onClick) {
-                        await onClick(event);
+                    if (!isLogin) {
+                        event.preventDefault();
+                        if (source === Source.Lens) await getWalletClientRequired(config);
```
Comment (guanbinrui) [inline src/components/MoreActionMenu.tsx:41]:
```suggestion
```

File: src/components/MoreActionMenu.tsx [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
 import { classNames } from '@/helpers/classNames.js';
+import { getWalletClientRequired } from '@/helpers/getWalletClientRequired.js';
+import { useIsLogin } from '@/hooks/useIsLogin.js';
+import { LoginModalRef } from '@/modals/controls.js';
 
-interface MoreActionMenuProps extends Omit<MenuProps<'div'>, 'onClick'> {
+interface MoreActionMenuProps extends MenuProps<'div'> {
     button: React.ReactNode;
     children: React.ReactNode;
+    source?: SocialSource;
```
Comment (guanbinrui) [inline src/components/MoreActionMenu.tsx:15]:
```suggestion
```

Comment (dudu0506) [reply src/components/MoreActionMenu.tsx:13]: we need this to open expected login modal

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**